cpp/src/arrow/array-dict-test.cc

```diff
@@ -740,4 +743,63 @@ TEST(TestDictionary, FromArray) {
   ASSERT_RAISES(Invalid, DictionaryArray::FromArrays(dict_type, indices4, &arr4));
 }
 
+TEST(TestDictionary, TransposeBasic) {
+  std::shared_ptr<Array> arr, out, expected;
```

Comment:
Can you type them `DictionaryArray` at the declaration to avoid all casts?

Reply:
Well, other APIs take `Array` pointers, not `DictionaryArray`.